tracing/internal/notracingtest/no_tracing_test.go

```diff
@@ -0,0 +1,105 @@
+package notracingtest
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel"
+
+	"github.com/grafana/dskit/tracing"
+)
+
+func TestNoTracingConfigured(t *testing.T) {
+	// Make sure no tracing env is set for the test.
+	defer unsetAndRestoreDeferred(
+		"JAEGER_AGENT_HOST",
+		"JAEGER_ENDPOINT",
+		"JAEGER_SAMPLER_MANAGER_HOST_PORT",
+		"OTEL_TRACES_EXPORTER",
+		"OTEL_EXPORTER_OTLP_ENDPOINT",
+		"OTEL_EXPORTER_OTLP_TRACES_ENDPOINT",
+	)()
+	logger := make(mockLogger, 100)
+
+	t.Run("tracing.NewOTelOrJaegerFromEnv", func(t *testing.T) {
+		closer, err := tracing.NewOTelOrJaegerFromEnv("test", logger)
+		require.NoError(t, err)
+		t.Cleanup(func() { require.NoError(t, closer.Close()) })
+
+		ensureTracingDoesNotLogErrors(t, logger)
+	})
+	t.Run("tracing.NewOTelFromEnv", func(t *testing.T) {
+		closer, err := tracing.NewOTelFromEnv("test", logger)
+		require.NoError(t, err)
+		t.Cleanup(func() { require.NoError(t, closer.Close()) })
+
+		ensureTracingDoesNotLogErrors(t, logger)
+	})
+}
+
+func ensureTracingDoesNotLogErrors(t *testing.T, logger mockLogger) {
+	tracer := otel.Tracer("test")
+	done := make(chan struct{})
+	defer close(done)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				// Generate A LOT of spans, so exporter will be forced to export them if configured.
+				_, sp := tracer.Start(context.Background(), "test-operation")
+				sp.End()
+			}
+		}
+	}()
+
+	// There's no way to figure out whether OTel tracer provider is configured or not, so we'll just observe the logs.
+	timeout := time.After(time.Second)
```

Comment:
Isn't a second a bit too much? If you undo your changes, how long does it take for this test to fail?